src/main/java/duke/task/Task.java

```diff
@@ -0,0 +1,87 @@
+package duke.task;
+
+/**
+ * Represents an abstract Task object. Inherited by Deadline, Event and Todo classes.
+ */
+public abstract class Task {
+    protected String description;
+    protected Boolean isDone;
+
+    private enum TaskType {
+        DEADLINE("deadline <task description> /by <yyyy-mm-dd>"),
+        EVENT("event <event description> /at <event location>"),
+        TODO("todo <task description>");
+
+        private final String format;
+
+        TaskType(String format) {
+            this.format = format;
+        }
+
+        String getFormat() {
+            return this.format;
+        }
+    }
+
+    protected Task(String description, boolean isDone) {
+        this.description = description;
+        this.isDone = isDone;
+    }
+
+    /**
+     * Returns format of command to add a specific task type.
+     * @param taskTypeString String indicating task type (event, deadline or todo).
+     * @return Format of command to add task of type as specified by taskTypeString.
+     */
+    public static String getFormat(String taskTypeString) {
+        return TaskType.valueOf(taskTypeString.toUpperCase()).getFormat();
+    }
+
+    /**
+     * Returns task in String format that is suitable for saving to and retrieving from hard disk.
+     * @return String of task in format for saving to and retrieving from hard disk.
+     */
+    public abstract String getParsedTask();
+
+    /**
+     * Returns checkbox with tick/cross representing if task is done (tick) or not (cross).
+     * @return String representing checkbox with unicode character CHECK MARK (U+2713) if task is done
+     * or HEAVY BALLOT X (U+2718) if task is not done.
+     */
+    public String getCheckBox() {
+        if (this.isDone) {
+            return "[\u2713]";
+        } else {
+            return "[\u2718]";
+        }
+    }
+
+    /**
+     * Marks this task as done.
+     */
+    public void markDone() {
```

Comment:
Should this be method be called `setDone` instead? See the boolean section in [java coding standard](https://se-education.org/guides/conventions/java/index.html#naming)

Reply:
Ah, I hesitated naming it setDone as it only sets done to true and not false (which does not seem to be the convention for a setter). I've renamed it to setDoneAsTrue ! :)